app/models/concerns/hostext/search.rb

```diff
@@ -177,6 +178,13 @@ def search_by_config_group(key, operator, value)
         {:conditions => opts, :include => :hostgroup}
       end
 
+      def search_by_proxy(key, operator, value)
+        conditions  = sanitize_sql_for_conditions(["smart_proxies.name #{operator} ?", value_to_sql(operator, value)])
+        host_ids = SmartProxy.where(conditions).flat_map{|proxy| proxy.all_hosts.pluck('hosts.id')}.uniq.join(',')
+        host_ids = '-1' if host_ids.empty?
+        {:conditions => "hosts.id IN (#{host_ids})"}
```

Comment:
Why not perform the joins to subnets, domains etc in the search query here?  You could avoid the separate lookup entirely (I see your response about the performance, but don't follow why this is preferable.)